File: aura-backend/server/models/SupportConversation.js
const mongoose = require('mongoose');

const supportConversationSchema = new mongoose.Schema(
  {
    user: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
      unique: true,
      index: true,
    },
    assignedAdmin: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Admin',
      default: null,
    },
    archived: {
      type: Boolean,
      default: false,
      index: true,
    },
    lastMessagePreview: {
      type: String,
      default: '',
      trim: true,
    },
    lastMessageBy: {
      type: String,
      enum: ['user', 'admin'],
      default: 'user',
    },
    lastMessageAt: {
      type: Date,
      default: null,
      index: true,
    },
    unreadForAdmin: {
      type: Number,
      default: 0,
      min: 0,
    },
    unreadForUser: {
      type: Number,
      default: 0,
      min: 0,
    },
  },
  { timestamps: true, strict: true }
);

supportConversationSchema.index({ archived: 1, lastMessageAt: -1, updatedAt: -1 });

module.exports = mongoose.model('SupportConversation', supportConversationSchema);

File: app/src/main/java/com/aurapc/admin/data/model/SupportConversation.java
package com.aurapc.admin.data.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class SupportConversation implements Serializable {

    @SerializedName(value = "_id", alternate = {"id"})
    public String id;

    @SerializedName("user")
    public Object user;

    @SerializedName("assignedAdmin")
    public Object assignedAdmin;

    @SerializedName("archived")
    public Boolean archived;

    @SerializedName("lastMessagePreview")
    public String lastMessagePreview;

    @SerializedName("lastMessageAt")
    public String lastMessageAt;

    @SerializedName("lastMessageBy")
    public String lastMessageBy;

    @SerializedName("unreadForAdmin")
    public Integer unreadForAdmin;

    @SerializedName("unreadForUser")
    public Integer unreadForUser;

    @SerializedName("createdAt")
    public String createdAt;

    @SerializedName("updatedAt")
    public String updatedAt;

    public String userName() {
        if (user instanceof java.util.Map) {
            Object phone = ((java.util.Map<?, ?>) user).get("phoneNumber");
            if (phone != null) return String.valueOf(phone);
            Object email = ((java.util.Map<?, ?>) user).get("email");
            if (email != null) return String.valueOf(email);
            Object profile = ((java.util.Map<?, ?>) user).get("profile");
            if (profile instanceof java.util.Map) {
                Object fullName = ((java.util.Map<?, ?>) profile).get("fullName");
                if (fullName != null) return String.valueOf(fullName);
            }
        }
        return "Khách hàng";
    }

    public String adminName() {
        if (assignedAdmin instanceof java.util.Map) {
            Object name = ((java.util.Map<?, ?>) assignedAdmin).get("name");
            if (name != null) return String.valueOf(name);
            Object email = ((java.util.Map<?, ?>) assignedAdmin).get("email");
            if (email != null) return String.valueOf(email);
        }
        return null;
    }
}